router/underlayproviders/udpip/udpip.go

```diff
@@ -363,61 +347,100 @@ func (u *udpConnection) send(batchSize int, pool chan *router.Packet) {
 	}
 }
 
+// makeHashSeed creates a new random number to serve as hash seed.
+// Each receive loop is associated with its own hash seed to compute
+// the proc queue where a packet should be delivered. All links that share
+// an underlying connection (therefore a receive loop) use the same hash seed.
+func makeHashSeed() uint32 {
+	hashSeed := fnv1aOffset32
+	randomBytes := make([]byte, 4)
+	if _, err := rand.Read(randomBytes); err != nil {
+		panic("Error while generating random value")
+	}
+	for _, c := range randomBytes {
+		hashSeed = hashFNV1a(hashSeed, c)
+	}
+	return hashSeed
+}
+
 // TODO(jiceatscion): use more inheritance between implementations?
 
 type externalLink struct {
-	queue      chan<- *router.Packet
+	egressQ    chan<- *router.Packet
+	procQs     []chan *router.Packet
+	metrics    router.InterfaceMetrics
 	ifID       uint16
-	bfdSession router.BFDSession
-	remote     netip.AddrPort
+	bfdSession *bfd.Session
+	pool       chan *router.Packet
+	seed       uint32
```

Comment:
The following layout would reduce memory usage from 80 to 72 bytes. However, I don't know whether the layout change could actually cause performance to drop due to cache alignment.

```suggestion
	metrics    router.InterfaceMetrics
	procQs     []chan *router.Packet
	egressQ    chan<- *router.Packet
	pool       chan *router.Packet
	bfdSession *bfd.Session
	seed       uint32
	ifID       uint16
```

Reply:
done. Might as well. Used a slightly different sorting I prefer procQs first, just in case the compiler can take advantage of the 0 offset. Same benefit (actually I got 64). Also sorted the other two to be consistent.
<!-- Reviewable comment -OL4WgbB7dw2YzGHYG5s:b-hxe3jv -->
<!-- Sent from Reviewable.io -->